utils/utils.go

```diff
@@ -680,7 +681,7 @@ func CustomizeHPA(hpa *autoscalingv1.HorizontalPodAutoscaler, ba common.BaseComp
 	hpa.Spec.ScaleTargetRef.Name = obj.GetName()
 	hpa.Spec.ScaleTargetRef.APIVersion = "apps/v1"
 
-	if ba.GetStorage() != nil {
+	if ba.GetStatefulSet() != nil && ba.GetStatefulSet().GetStorage() != nil {
```

Comment:
This should check only for `ba.GetStatefulSet() != nil` - because StatefulSet will be created for that scenario (whether storage is configured or not).